lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -5544,7 +5538,31 @@ void ModuleEmitter::emitHWModule(HWModuleOp module) {
   // If we have any parameters, print them on their own line.
   emitParameters(module, module.getParameters());
 
-  emitPortList(module, module.getPorts());
+  emitPortList(module, module.getPortList());
+
+  assert(state.pendingNewline);
+
+  // Emit the body of the module.
+  StmtEmitter(*this).emitStatementBlock(*module.getBodyBlock());
+  startStatement();
+  ps << "endmodule" << PP::newline;
+  setPendingNewline();
+
+  currentModuleOp = nullptr;
+}
+
+void ModuleEmitter::emitHWTestModule(HWTestModuleOp module) {
+  currentModuleOp = module;
+
+  emitComment(module.getCommentAttr());
+  emitSVAttributes(module);
+  startStatement();
+  ps << "module " << PPExtString(getVerilogModuleName(module));
+
+  // If we have any parameters, print them on their own line.
+  emitParameters(module, module.getParameters());
+
+  emitPortList(module, module.getPortList());
```

Comment:
With the port list simplified we might even be able to move some of these into a custom parser/printer directive inside the regular assembly format.